lib/token-types/bearer-token-type.js

```diff
@@ -0,0 +1,52 @@
+
+/**
+ * Module dependencies.
+ */
+
+var InvalidArgumentError = require('../errors/invalid-argument-error');
+
+/**
+ * Constructor.
+ */
+
+function BearerTokenType(accessToken, accessTokenLifetime, refreshToken, scope) {
+  if (!accessToken) {
+    throw new InvalidArgumentError('Missing parameter: `accessToken`');
+  }
+
+  this.accessToken = accessToken;
+  this.accessTokenLifetime = accessTokenLifetime;
+  this.refreshToken = refreshToken;
+  this.scope = scope;
```

Comment:
What about just allowing for optional properties?  The [spec envisions this](https://tools.ietf.org/html/rfc6749#section-5.1):  

```
 An example successful response:

     HTTP/1.1 200 OK
     Content-Type: application/json;charset=UTF-8
     Cache-Control: no-store
     Pragma: no-cache

     {
       "access_token":"2YotnFZFEjr1zCsicMWpAA",
       "token_type":"example",
       "expires_in":3600,
       "refresh_token":"tGzv3JOkF0XG5Qx2TlKWIA",
       "example_parameter":"example_value"
     }
```